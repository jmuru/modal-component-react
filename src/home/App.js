import React, { Component } from "react";
import Modal from "../modal/modal";
import "./App.css";

class App extends Component {
	constructor(props) {
		super(props);
		this.state = {
			modalBaseActive: false,
			modalWarningActive: false
		};
	}

	toggleModalBase = () => {
		this.setState({
			modalBaseActive: !this.state.modalBaseActive
		});
	};

	toggleModalWarning = () => {
		this.setState({
			modalWarningActive: !this.state.modalWarningActive
		});
	};

	render() {
		return (
			<div className="App">
				<h1>Modal Example</h1>
				<div>
					<h2>Base Modal</h2>
					<button className="modal-base-toggle" onClick={this.toggleModalBase}>
						this will activate base modal
					</button>
					{this.state.modalBaseActive ? (
						<Modal
							title={"test modal"}
							subtitle={"test subtitle"}
							modalBody={<div>hello world</div>}
							closeModal={this.toggleModalBase}
						/>
					) : null}
				</div>
				<div>
					<h2>Warning Modal</h2>
					<button
						className="modal-warning-toggle"
						onClick={this.toggleModalWarning}
					>
						this will activate warning modal
					</button>
					{this.state.modalWarningActive ? (
						<Modal
							title={"Warning Modal"}
							subtitle={"warning subtitle"}
							modalBody={<div>this is a warning</div>}
							modalType={"warning"}
							closeModal={this.toggleModalWarning}
						/>
					) : null}
				</div>
			</div>
		);
	}
}

export default App;
