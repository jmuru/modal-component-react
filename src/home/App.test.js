import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme,{ shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from './App';

Enzyme.configure({adapter: new Adapter()});

describe('Homepage', () => {

  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('should match snapshot when modal closed', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should matchsnapshot when modal open', () => {
    const wrapper = shallow(<App />);
    wrapper.find('.modal-base-toggle').simulate('click');
    expect(wrapper).toMatchSnapshot();
  });

})


