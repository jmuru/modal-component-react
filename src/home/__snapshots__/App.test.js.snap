// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage should match snapshot when modal closed 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Modal Example
</h1>,
        <div>
          <h2>
                    Base Modal
          </h2>
          <button
                    className="modal-base-toggle"
                    onClick={[Function]}
          >
                    this will activate base modal
          </button>
</div>,
        <div>
          <h2>
                    Warning Modal
          </h2>
          <button
                    className="modal-warning-toggle"
                    onClick={[Function]}
          >
                    this will activate warning modal
          </button>
</div>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Modal Example",
        },
        "ref": null,
        "rendered": "Modal Example",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Base Modal
</h2>,
            <button
              className="modal-base-toggle"
              onClick={[Function]}
>
              this will activate base modal
</button>,
            null,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Base Modal",
            },
            "ref": null,
            "rendered": "Base Modal",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "this will activate base modal",
              "className": "modal-base-toggle",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "this will activate base modal",
            "type": "button",
          },
          null,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Warning Modal
</h2>,
            <button
              className="modal-warning-toggle"
              onClick={[Function]}
>
              this will activate warning modal
</button>,
            null,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Warning Modal",
            },
            "ref": null,
            "rendered": "Warning Modal",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "this will activate warning modal",
              "className": "modal-warning-toggle",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "this will activate warning modal",
            "type": "button",
          },
          null,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Modal Example
</h1>,
          <div>
            <h2>
                        Base Modal
            </h2>
            <button
                        className="modal-base-toggle"
                        onClick={[Function]}
            >
                        this will activate base modal
            </button>
</div>,
          <div>
            <h2>
                        Warning Modal
            </h2>
            <button
                        className="modal-warning-toggle"
                        onClick={[Function]}
            >
                        this will activate warning modal
            </button>
</div>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Modal Example",
          },
          "ref": null,
          "rendered": "Modal Example",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Base Modal
</h2>,
              <button
                className="modal-base-toggle"
                onClick={[Function]}
>
                this will activate base modal
</button>,
              null,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Base Modal",
              },
              "ref": null,
              "rendered": "Base Modal",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "this will activate base modal",
                "className": "modal-base-toggle",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "this will activate base modal",
              "type": "button",
            },
            null,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Warning Modal
</h2>,
              <button
                className="modal-warning-toggle"
                onClick={[Function]}
>
                this will activate warning modal
</button>,
              null,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Warning Modal",
              },
              "ref": null,
              "rendered": "Warning Modal",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "this will activate warning modal",
                "className": "modal-warning-toggle",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "this will activate warning modal",
              "type": "button",
            },
            null,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Homepage should matchsnapshot when modal open 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Modal Example
</h1>,
        <div>
          <h2>
                    Base Modal
          </h2>
          <button
                    className="modal-base-toggle"
                    onClick={[Function]}
          >
                    this will activate base modal
          </button>
          <Modal
                    closeModal={[Function]}
                    modalBody={
                              <div>
                                        hello world
                              </div>
                    }
                    subtitle="test subtitle"
                    title="test modal"
          />
</div>,
        <div>
          <h2>
                    Warning Modal
          </h2>
          <button
                    className="modal-warning-toggle"
                    onClick={[Function]}
          >
                    this will activate warning modal
          </button>
</div>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Modal Example",
        },
        "ref": null,
        "rendered": "Modal Example",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Base Modal
</h2>,
            <button
              className="modal-base-toggle"
              onClick={[Function]}
>
              this will activate base modal
</button>,
            <Modal
              closeModal={[Function]}
              modalBody={
                            <div>
                                          hello world
                            </div>
              }
              subtitle="test subtitle"
              title="test modal"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Base Modal",
            },
            "ref": null,
            "rendered": "Base Modal",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "this will activate base modal",
              "className": "modal-base-toggle",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "this will activate base modal",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "closeModal": [Function],
              "modalBody": <div>
                hello world
</div>,
              "subtitle": "test subtitle",
              "title": "test modal",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Warning Modal
</h2>,
            <button
              className="modal-warning-toggle"
              onClick={[Function]}
>
              this will activate warning modal
</button>,
            null,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Warning Modal",
            },
            "ref": null,
            "rendered": "Warning Modal",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "this will activate warning modal",
              "className": "modal-warning-toggle",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "this will activate warning modal",
            "type": "button",
          },
          null,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Modal Example
</h1>,
          <div>
            <h2>
                        Base Modal
            </h2>
            <button
                        className="modal-base-toggle"
                        onClick={[Function]}
            >
                        this will activate base modal
            </button>
            <Modal
                        closeModal={[Function]}
                        modalBody={
                                    <div>
                                                hello world
                                    </div>
                        }
                        subtitle="test subtitle"
                        title="test modal"
            />
</div>,
          <div>
            <h2>
                        Warning Modal
            </h2>
            <button
                        className="modal-warning-toggle"
                        onClick={[Function]}
            >
                        this will activate warning modal
            </button>
</div>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Modal Example",
          },
          "ref": null,
          "rendered": "Modal Example",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Base Modal
</h2>,
              <button
                className="modal-base-toggle"
                onClick={[Function]}
>
                this will activate base modal
</button>,
              <Modal
                closeModal={[Function]}
                modalBody={
                                <div>
                                                hello world
                                </div>
                }
                subtitle="test subtitle"
                title="test modal"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Base Modal",
              },
              "ref": null,
              "rendered": "Base Modal",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "this will activate base modal",
                "className": "modal-base-toggle",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "this will activate base modal",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "closeModal": [Function],
                "modalBody": <div>
                  hello world
</div>,
                "subtitle": "test subtitle",
                "title": "test modal",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Warning Modal
</h2>,
              <button
                className="modal-warning-toggle"
                onClick={[Function]}
>
                this will activate warning modal
</button>,
              null,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Warning Modal",
              },
              "ref": null,
              "rendered": "Warning Modal",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "this will activate warning modal",
                "className": "modal-warning-toggle",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "this will activate warning modal",
              "type": "button",
            },
            null,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
